zerver/views/users.py

```diff
@@ -114,6 +115,19 @@ def update_user_backend(request, user_profile, email,
 
     return json_success()
 
+@has_request_variables
+def update_buddy_list(request, user_profile, buddy_id=REQ(validator=check_string),
+                      should_add=REQ(validator=check_string)):
+    # type: (HttpRequest, UserProfile, str, str) -> HttpResponse
+    try:
+        buddy_profile = get_user_profile_by_id(buddy_id)
+    except UserProfile.DoesNotExist:
+        return json_error(_('No such user with user id ' + buddy_id))
```

Comment:
We probably want to add two more validations here, and appropriate tests:
* your "buddy" must not be yourself
* your "buddy" must be in your realm

I'm trying to think of other validations.  It would be kind of strange to be buddy with a "bot," but I don't think it should be illegal.